src/map/battle.c

```diff
@@ -3176,6 +3206,19 @@ int64 battle_calc_damage(struct block_list *src,struct block_list *bl,struct Dam
 				status_change_end(bl, SC_KYRIE, INVALID_TIMER);
 		}
 
+		if ((sce = sc->data[SC_TUNAPARTY]) != NULL && damage > 0) {
+			clif->specialeffect(bl, 336, AREA);
+			sce->val2 -= (int)cap_value(damage, INT_MIN, INT_MAX);
+			if (sce->val2 >= 0) {
+				damage = 0;
+			} else {
+				damage = -sce->val2;
+			}
+			if (/*(--sce->val3) <= 0 ||*/ (sce->val2 <= 0)) {
```

Comment:
/*(--sce->val3) <= 0 ||*/ it would be better if this comment out of the if with explain of it.